indexFormat/implementation.md

```diff
@@ -1,77 +1,67 @@
+> Are we missing an implementation? File a new issue on GitHub to add it here.
 
-## LSIF implementation skeleton
+## LSIF exporter skeleton
 
-As detailed in the specification, the LSIF exporter consists of two tools: the index exporter and the package manager linker.
+As [detailed in the spec](specification.md#project-exports-and-external-imports), the LSIF exporter consists of two tools: the index exporter and the package linker.
 
 ### Index exporter
 
 This tool generates an LSIF dump for a workspace by traversing through source files and storing LSP responses. For TypeScript/JavaScript, [`lsif-tsc`](https://github.com/Microsoft/lsif-node/tree/master/tsc) is the index exporter.
 
-### Package manager linker
+### Package linker
 
-This tool ingests the LSIF output from an index exporter and converts `moniker` vertices into the package manager scheme. Additionally, it links these monikers to `packageInformation` vertices. For TypeScript/JavaScript, [`lsif-npm`](https://github.com/Microsoft/lsif-node/tree/master/npm) is the package manager linker for NPM.
+This tool ingests the LSIF output from an index exporter and converts `moniker` vertices into the package manager scheme. Additionally, it links these monikers to `packageInformation` vertices that reference packages hosted on a registry. For TypeScript/JavaScript, [`lsif-npm`](https://github.com/Microsoft/lsif-node/tree/master/npm) is the package manager linker for NPM.
 
 ## Testing and validation
 
-### lsif-util
+### LSIF validation utility
 
-The `lsif-util` tool can validate your generated LSIF output. Additionally, the tool can also help you comprehend the generated output by visualizing and searching the vertices and edges.
+The [`lsif-util`](https://github.com/jumattos/lsif-util) tool can validate your generated LSIF output. Additionally, the tool can also help you comprehend the generated output by visualizing and searching the vertices and edges.
 
 ### VS Code LSIF extension
 
-With the [LSIF extension for VS Code](https://github.com/Microsoft/vscode-lsif-extension), you can power navigation in the editor over a generated LSIF file, instead of running a language server.
+With the [LSIF extension for VS Code](https://github.com/Microsoft/vscode-lsif-extension), you can dogfood an LSIF index to power navigation inside VS Code.
 
 ## Recommended patterns
 
-### Execution from CLI
-
-asdf
-
-### `--outputFormat` flag
-
-1. `json`: Output is a JSON array, with each JSON object (vertex or edge) as an element of this array.
-1. `line`: Output is a series of lines, each line is a vertex or an edge. This format is more friendly to streaming output, and therefore preferred.
-
-## `--projectRoot flag`
+We have seen the following patterns work well in existing implementations.
 
-asdf
+### Cross-platform
 
-## Packaging
+If the LSIF exporter does not work across platforms (Windows, Linux, Mac), platform dependencies should be called out.
 
-## Checklist
+### `--outputFormat` execution flag
```

Comment:
Another good option that @jdneo brought up is a verbose flag so we don't have to process any excess output that isn't json when streaming the content. This isn't necessary, just recommended.

Reply:
Interesting. I'm going to leave this out for now, since debugging/verbosity implementation can be left to the implementers. Will keep in feedback for subsequent iterations.